### PR TITLE
Making perturbation alpha into list input and bugfixes

### PR DESCRIPTION
**Pull request by WPringle**
See issues #35 #36 

- [x] Allowing for alphas to be a list input to perturbation/atcf.py so that each ensemble member has a different perturbation of the error
- [x] In the best track perturbation test making a separate call to perturb the radius_of_maximum_winds since that has a different alpha range
- [x] Trying to fix bug for negative along_track perturbation due to variable index range, still having issues need figure out
- [x] Noting problem with radius_maximum_winds perturbation always ending up as 200 nm, need to fix

This requires more fixes to come for the AlongTrack and RadiusOfMaxmimumWinds perturbations.

### CODE REVIEW

**Comment by WPringle** [comment]
Cross-track example perturbed tracks with test from this PR (make sense?):
cross_track_compare

**Comment by WPringle** [comment]
Here are the reference figures for the test we want to do (alpha = [-1,1] for track and Vmax; and [0.25,0.75] for Rmax) using the original `make_storm_ensemble.py` script from the [Storm-Surge-Mesh-Subetting](https://github.com/WPringle/Storm-Surge-Mesh-Subsetting) repository:
cross_track_compare
along_track_compare
radius_of_maximum_winds_compare
max_sustained_wind_speed_compare

**Comment by WPringle** [comment]
Cross-track example perturbed tracks with test from this PR, quite different from the original one.. :
cross_track_compare

**Comment by WPringle** [comment]
@zacharyburnettNOAA could there be some issue with the units or something like that?

**Comment by WPringle** [comment]
The wind speed one looks fine for the current PR:
max_sustained_wind_speed_compare

**Comment by WPringle** [comment]
New result for Rmax after correction looks good:
radius_of_maximum_winds_compare

**Comment by WPringle** [comment]
Found mistake in the cross_track and along_track errors, I never entered them in correctly, and instead just copied them from the intensity (Vmax) errors. Here are new plots for cross-track and along-track perturbations using the Storm-Surge-Mesh-Subsetting repository:
cross_track_compare
along_track_compare

**Comment by zacharyburnettNOAA** [comment]
@WPringle sorry I did not see this on Friday. It looks like I accidentally assigned `.us_statute_mile` instead of `.nautical_mile` in the errors; which unit are the values in?

**Comment by WPringle** [comment]
> @WPringle sorry I did not see this on Friday. It looks like I accidentally assigned `.us_statute_mile` instead of `.nautical_mile` in the `historical_forecast_errors`; which unit are the values in?
> 
> https://github.com/noaa-ocs-modeling/EnsemblePerturbation/blob/d99a723f4a9dabb3ba3e5f63cf5c99b899657839/ensembleperturbation/perturbation/atcf.py#L410-L414

No, this is fine actually thanks! The values in the array are statute miles here but we want them to be converted to nautical miles for use in the vortex file. (we could just manually compute these values in nautical miles to avoid confusion).

The problem was the dataframe was just populating nans due to the format of how the values were entered.

**Comment by zacharyburnettNOAA** [comment]
tests look okay and the tracks are now being correctly perturbed as shown in your diagrams; I will merge this

File: tests/test_track_ensemble.py [hunk 1]
```diff
@@ -18,21 +18,28 @@ def test_besttrack_ensemble():
...
-    variables = [
-        MaximumSustainedWindSpeed,
-        RadiusOfMaximumWinds,
-        AlongTrack,
-        CrossTrack,
-    ]
-
+    # list of variables where perturbation is Gaussian
+    gauss_variables = [MaximumSustainedWindSpeed, CrossTrack]
+    # AlongTrack,
```
Comment (zacharyburnettNOAA) [inline tests/test_track_ensemble.py:23]: did you not want to test the along track as well?

**Comment by WPringle** [comment]
Can please wait, I need to go through the tracks perturbation. The plots above were not for this script (for the old one). I'm happy with the Vmax and Rmax for this script but I think there is still an issue with the cross-track/along-track. Will try to get it sorted this afternoon.

**Review by zacharyburnettNOAA — COMMENTED** [review]
> Can please wait, I need to go through the tracks perturbation. The plots above were not for this script (for the old one). I'm happy with the Vmax and Rmax for this script but I think there is still an issue with the cross-track/along-track. Will try to get it sorted this afternoon.

Sure thing, I will wait. It looks like the tests aren't checking the reference directory anyway so lemme know when youre ready

**Comment by zacharyburnettNOAA** [comment]
I am having some difficulty understanding the along track perturbation methodology; would it be okay if we had a google meet or something to talk about it?

**Comment by WPringle** [comment]
Yeah I'm free today if you are?

**Comment by zacharyburnettNOAA** [comment]
> Yeah I'm free today if you are?

alright, I made a meeting at https://meet.google.com/ffc-tjbo-dic

we can also meet later this week if youre available, it's not super urgent

**Comment by WPringle** [comment]
Ok now the `cross-track` is behaving as expected; the order of longitude and latitude was switched. Last thing is to figure out the `along-track`

**Comment by WPringle** [comment]
So the new code for `along_track` interpolation seems to work ok now, but the extrapolation at the beginning is strange. It should look like the black line but its being extrapolated as the red-line (from where the black line starts). 
image

**Comment by zacharyburnettNOAA** [comment]
> @zacharyburnettNOAA So the new code for `along_track` interpolation seems to work ok now, but the extrapolation at the beginning is strange. It should look like the black line but its being extrapolated as the red-line (from where the black line starts). I don't quite get the extrapolation as you did it but should be easy fix for the sign.
> image

looks like this was because the `numpy.flip` function was flipping both axes (because I forgot to specify `axis=0`), fixed in 455d43f

**Comment by zacharyburnettNOAA** [comment]
@WPringle from what I can see, this seems pretty good. Let me know if there are any more outstanding issues aside from the 4 you specified at the beginning

**Review by zacharyburnettNOAA — APPROVED** [review]

**Pull request merged**